File: src/dataset/python/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_triangle_v1.py
"""Validate and specify triangle types"""

class TriangleError(Exception):
    """Error if it cannot be a triangle"""
    pass

class Triangle:
    """Validate and specify triangle types"""
    __slots__ = ('a', 'b', 'c')
    
    def __init__(self, a, b, c):
        """Create and validate triangle"""
        sides = (a, b, c)
        min_side = min(sides)
        max_side = max(sides)
        mid_side = a + b + c - min_side - max_side
        
        if min_side + mid_side <= max_side:
            raise TriangleError("Not a Triangle")
        
        self.a, self.b, self.c = min_side, mid_side, max_side

    def kind(self):
        """Specify triangle type"""
        if self.a == self.c:
            return "equilateral"
        elif self.a == self.b or self.b == self.c:
            return "isosceles"
        else:
            return "scalene"

File: src/dataset/Java/triangle_exercism-ThomasZumsteg/claude/ClaudeSonnet4_Triangle_v1.java
public class Triangle {
    private double a, b, c;

    public Triangle(double a, double b, double c) throws TriangleException {
        // Sort manually without creating array or using Arrays.sort
        double min, mid, max;
        if (a <= b && a <= c) {
            min = a;
            if (b <= c) {
                mid = b;
                max = c;
            } else {
                mid = c;
                max = b;
            }
        } else if (b <= a && b <= c) {
            min = b;
            if (a <= c) {
                mid = a;
                max = c;
            } else {
                mid = c;
                max = a;
            }
        } else {
            min = c;
            if (a <= b) {
                mid = a;
                max = b;
            } else {
                mid = b;
                max = a;
            }
        }
        
        this.a = min;
        this.b = mid;
        this.c = max;
        
        if (min < 0 || min + mid <= max) {
            throw new TriangleException();
        }
    }

    public TriangleKind getKind() {
        if (a == b) {
            return (b == c) ? TriangleKind.EQUILATERAL : TriangleKind.ISOSCELES;
        } else if (b == c) {
            return TriangleKind.ISOSCELES;
        } else {
            return TriangleKind.SCALENE;
        }
    }
}